yaksh/views.py

```diff
@@ -954,3 +963,71 @@ def design_questionpaper(request):
         context = {'form': form, 'questionpaper':True}
         return my_render_to_response('yaksh/design_questionpaper.html',
                                      context, context_instance=ci)
+
+
+@login_required
+def download_questions(request):
+    user = request.user
+    if not is_moderator(user):
+        raise Http404('You are not allowed to view this page!')
+    questions = Question.objects.filter(user_id = user.id)
+    response = HttpResponse(content_type='text/csv')
+    response['Content-Disposition'] = 'attachment; filename="{0}_Questions.csv"'.format(user)
+    writer = csv.writer(response)
+    header = [
+                'summary',
+                'description',
+                'points',
+                'test',
+                'ref_code_path',
+                'options',
+                'language',
+                'type',
+                'active',
+                'snippet'
+    ]
+    writer.writerow(header)
+    for que in questions:
+        row = [
+                que.summary,
+                que.description,
+                que.points,
+                que.test,
+                que.ref_code_path,
+                que.options,
+                que.language,
+                que.type,
+                que.active,
+                que.snippet
+        ]
+        writer.writerow(row)
+    return response
+
+
+@login_required
+def upload_questions(request):
+    user = request.user
+    ci = RequestContext(request)
+
+    if not is_moderator(user):
+        raise Http404('You are not allowed to view this page!')
+    if request.method == 'POST':
+        form = UploadFileForm(request.POST, request.FILES)
+        if form.is_valid():
+            csvfile = request.FILES['file']
+            if csvfile.name.split('.')[1] == "csv":
+                reader = csv.DictReader(csvfile, delimiter=',')
```

Comment:
This and the download should really be model methods and not methods on the view.